README.md
# MotorCortex-Openlayers

**Table of Contents**

- [MotorCortex-Openlayers](#motorcortex-openlayers)
  - [Demo](#demo)
- [Intro / Features](#intro--features)
- [Getting Started](#getting-started)
  - [Installation](#installation)
  - [Importing and Loading](#importing-and-loading)
- [Creating Incidents](#creating-incidents)
  - [Map Clip](#map-clip)
  - [GoTo](#goto)
- [Adding Incidents in your clip](#adding-incidents-in-your-clip)
- [Contributing](#contributing)
- [License](#license)
- [Sponsored by](#sponsored-by)

## Demo

[Check it out here](https://donkeyclip.github.io/motorcortex-ol/demo/)

# Intro / Features
MotorCortex Openlayers takes the capabilities of [Openlayers](https://openlayers.org/) library of creating a dynamic map in any web page.
The library exposes a Map Clip with the name Clip which will initialize an Openlayer Map instance where you can add animation with the "GoTo" Incident.

This Plugin exposes two Incident:
- Map Clip
- GoTo

# Getting Started

## Installation
```bash
$ npm install @donkeyclip/motorcortex-ol
# OR
$ yarn add @donkeyclip/motorcortex-ol
```
## Importing and loading

```javascript
import { loadPlugin } from "@donkeyclip/motorcortex";
import MapsDef from "@donkeyclip/motorcortex-ol";
const Maps = loadPlugin(MapsDef);
```

# Creating Incidents

## Map Clip
```javascript
const london = MapsDef.utils.fromLonLat([-0.12755, 51.507222]);
const bern = MapsDef.utils.fromLonLat([7.4458, 46.95]);

const clip = new Maps.Clip(
  {
    parameters: {
      view: { center: london, zoom: 8 }
    }
  },
  {
    host: document.getElementById("clip"),
    containerParams: { width: "1280px", height: "720px" }
  }
);
```
### Map Clip Attrs
Map Clip take as a parameter a `view` object. This object contains the starting point (`center`) and the `zoom` number.
The `center` value has the following structure: 
```javascript
center: MapsDef.utils.fromLonLat([-0.12755, 51.507222])
```

## GoTo
```javascript
const gotoBern = new Maps.GoTo(
  {
    animatedAttrs: {
      goto: {
        zoom: 3,
        center: bern
      }
    }
  },
  { duration: 4000, selector: "!#olmap" }
);
```
### GoTo Attrs 
Goto Incident take as an attribute a `goto` object.This object contains the ending point (`center`) and the `zoom` number.

#### IMPORTANT
Along with the attributes, all `GoTo incidents` must take on their props the `selector` key with the value: `!#olmap`

# Adding Incidents in your clip

```javascript
mapsClipName.addIncident(incidentGoToName,startTime);
```

# Contributing 

In general, we follow the "fork-and-pull" Git workflow, so if you want to submit patches and additions you should follow the next steps:
1.	**Fork** the repo on GitHub
2.	**Clone** the project to your own machine
3.	**Commit** changes to your own branch
4.	**Push** your work back up to your fork
5.	Submit a **Pull request** so that we can review your changes

# License

[MIT License](https://opensource.org/licenses/MIT)

# Sponsored by
[<img src="https://presskit.donkeyclip.com/logos/donkey%20clip%20logo.svg" width=250></img>](https://donkeyclip.com)


CHANGELOG.md
# Changelog

## [3.1.0](https://github.com/donkeyclip/motorcortex-ol/compare/v3.0.1...v3.1.0) (2022-04-07)


### Features

* drop explorer 11 support ([71cc889](https://github.com/donkeyclip/motorcortex-ol/commit/71cc889d2bcc0f457204cdc99b554a504e530d53))
* update compatibility with motorcortex ([499e125](https://github.com/donkeyclip/motorcortex-ol/commit/499e125d67df309a789c2c1dadf63e440de13cab))

### [3.0.1](https://github.com/donkeyclip/motorcortex-ol/compare/v3.0.0...v3.0.1) (2022-03-25)


### Bug Fixes

* upgrade mc to latest version ([e10d27c](https://github.com/donkeyclip/motorcortex-ol/commit/e10d27cc915438e1f06851507df52a0aaa236d73))

## [3.0.0](https://github.com/donkeyclip/motorcortex-ol/compare/v2.0.8...v3.0.0) (2022-03-24)


### ⚠ BREAKING CHANGES

* update mc version

### Features

* finish ol integration ([5fae6cc](https://github.com/donkeyclip/motorcortex-ol/commit/5fae6cca35bb3c74b163b92794c53516be699754))
* fix bug at context handler ([5438637](https://github.com/donkeyclip/motorcortex-ol/commit/543863782240252b919fa91918956c66440bab2d))
* fix new map clip ([8232cfe](https://github.com/donkeyclip/motorcortex-ol/commit/8232cfe13df7d240b7238472e681cf52184645ae))
* implement custom animate function for ol maps ([68fd3c6](https://github.com/donkeyclip/motorcortex-ol/commit/68fd3c69f04c54134ae1abde5a6a1f07a0fab478))
* implementation with openlayer 6 ([5e022ed](https://github.com/donkeyclip/motorcortex-ol/commit/5e022ed9faa14adf239936fb7de3f43e0c44caa9))
* initialize ([527bb8b](https://github.com/donkeyclip/motorcortex-ol/commit/527bb8baa219a6cfb941c6404bc4a30590c21e36))
* new version ([c1da7d1](https://github.com/donkeyclip/motorcortex-ol/commit/c1da7d182585a3d35ec387460d740913a3be198d))
* new version ([724fa5d](https://github.com/donkeyclip/motorcortex-ol/commit/724fa5d0de960016de35f98c76202ca6d30d3950))
* promise integration, ol integration ([d948e2e](https://github.com/donkeyclip/motorcortex-ol/commit/d948e2e2ebed6a5f23df7d554d2ba9359539e396))
* update mc version ([f70ac11](https://github.com/donkeyclip/motorcortex-ol/commit/f70ac1190b46baf94b6cd5efeba26721dda440d1))
* update to ol 6.13.0 ([b0c0d62](https://github.com/donkeyclip/motorcortex-ol/commit/b0c0d6271b82b6234ade1f0458397d434d4b47aa))
* updgrade to new mc version ([8c31c4f](https://github.com/donkeyclip/motorcortex-ol/commit/8c31c4f1f00e3ecfe951b151bda2f67b160a0120))


### Bug Fixes

* add json plugin to rollup ([14fc29c](https://github.com/donkeyclip/motorcortex-ol/commit/14fc29c6dfca7c134e2c0ab35a217de4d4aff543))
* add version and name at index.js ([ba8d241](https://github.com/donkeyclip/motorcortex-ol/commit/ba8d24155a3a138ca15fbd69a7c0e904d1802c27))
* change eslintingore ([117bf8a](https://github.com/donkeyclip/motorcortex-ol/commit/117bf8a9d4a70c1c0bf95fb22fac9ec71a6407f4))
* change zoom to function and improve readme ([dcd8f11](https://github.com/donkeyclip/motorcortex-ol/commit/dcd8f11d48d549942daa4a1a6305b20cf06b590e))
* fix eslint parser ([7d5e2e7](https://github.com/donkeyclip/motorcortex-ol/commit/7d5e2e799d9d933edea9676952955fa8249c220b))
* move monorepo to devdeps ([ffce040](https://github.com/donkeyclip/motorcortex-ol/commit/ffce0409ac055fb6d3d024c1fc3ece8ba4a7c351))
* preload more tiles for a smoother experience ([4dd5ce0](https://github.com/donkeyclip/motorcortex-ol/commit/4dd5ce0bade20c8abd7be42837f2e90c273fd09c))
* remove ol easings ([f995e74](https://github.com/donkeyclip/motorcortex-ol/commit/f995e74ef48104a56f392e46224d244dcf3cd22e))
* update mc and player ([c7878b3](https://github.com/donkeyclip/motorcortex-ol/commit/c7878b35c638a2e77f1e5b40c7afee2cc6343469))


### Miscellaneous Chores

* release 2.0.7 ([d3ed8a3](https://github.com/donkeyclip/motorcortex-ol/commit/d3ed8a3153699d6d74a963dd750aa389e77c0a9b))
* release 3.0.0 ([84b6bdf](https://github.com/donkeyclip/motorcortex-ol/commit/84b6bdfab44a14460a25c5a0129dfcaa03f3cc66))

### [2.0.8](https://www.github.com/donkeyclip/motorcortex-ol/compare/v2.0.7...v2.0.8) (2021-11-17)


### Bug Fixes

* add json plugin to rollup ([14fc29c](https://www.github.com/donkeyclip/motorcortex-ol/commit/14fc29c6dfca7c134e2c0ab35a217de4d4aff543))
* update mc and player ([c7878b3](https://www.github.com/donkeyclip/motorcortex-ol/commit/c7878b35c638a2e77f1e5b40c7afee2cc6343469))

### 2.0.7 (2021-04-16)


### Features

* finish ol integration ([5fae6cc](https://www.github.com/kissmybutton/motorcortex-ol/commit/5fae6cca35bb3c74b163b92794c53516be699754))
* fix bug at context handler ([5438637](https://www.github.com/kissmybutton/motorcortex-ol/commit/543863782240252b919fa91918956c66440bab2d))
* fix new map clip ([8232cfe](https://www.github.com/kissmybutton/motorcortex-ol/commit/8232cfe13df7d240b7238472e681cf52184645ae))
* implement custom animate function for ol maps ([68fd3c6](https://www.github.com/kissmybutton/motorcortex-ol/commit/68fd3c69f04c54134ae1abde5a6a1f07a0fab478))
* implementation with openlayer 6 ([5e022ed](https://www.github.com/kissmybutton/motorcortex-ol/commit/5e022ed9faa14adf239936fb7de3f43e0c44caa9))
* initialize ([527bb8b](https://www.github.com/kissmybutton/motorcortex-ol/commit/527bb8baa219a6cfb941c6404bc4a30590c21e36))
* new version ([c1da7d1](https://www.github.com/kissmybutton/motorcortex-ol/commit/c1da7d182585a3d35ec387460d740913a3be198d))
* new version ([724fa5d](https://www.github.com/kissmybutton/motorcortex-ol/commit/724fa5d0de960016de35f98c76202ca6d30d3950))
* promise integration, ol integration ([d948e2e](https://www.github.com/kissmybutton/motorcortex-ol/commit/d948e2e2ebed6a5f23df7d554d2ba9359539e396))
* updgrade to new mc version ([8c31c4f](https://www.github.com/kissmybutton/motorcortex-ol/commit/8c31c4f1f00e3ecfe951b151bda2f67b160a0120))


### Bug Fixes

* add version and name at index.js ([ba8d241](https://www.github.com/kissmybutton/motorcortex-ol/commit/ba8d24155a3a138ca15fbd69a7c0e904d1802c27))

demo/index.js
import { loadPlugin } from "@donkeyclip/motorcortex";
import Player from "@donkeyclip/motorcortex-player";
import MapsDef from "../src/";

const Maps = loadPlugin(MapsDef);

const london = MapsDef.utils.fromLonLat([-0.12755, 51.507222]);
const moscow = MapsDef.utils.fromLonLat([37.6178, 55.7517]);
const bern = MapsDef.utils.fromLonLat([7.4458, 46.95]);

const map = new Maps.Clip(
  {
    parameters: {
      view: { center: london, zoom: 8 },
    },
  },
  {
    host: document.getElementById("clip"),
    containerParams: { width: "1280px", height: "720px" },
  }
);

const gotoBern = new Maps.GoTo(
  {
    animatedAttrs: {
      goto: {
        zoom: 3,
        center: bern,
      },
    },
  },
  { duration: 4000, selector: "!#olmap", easing: "easeInExpo" }
);

const gotoMoscow = new Maps.GoTo(
  {
    animatedAttrs: {
      goto: {
        zoom: 8,
        center: moscow,
      },
    },
  },
  { duration: 4000, selector: "!#olmap", easing: "easeInExpo" }
);

map.addIncident(gotoBern, 0);
map.addIncident(gotoMoscow, 4000);
new Player({ clip: map });
